packages-web/maps/src/components/LeafletMap.tsx

```diff
@@ -0,0 +1,180 @@
+import { createElement, ReactElement, useEffect, useRef, useState } from "react";
+import {
+    FeatureGroup,
+    LatLngLiteral,
+    Map,
+    MapOptions,
+    Marker as LeafletMarker,
+    icon,
+    tileLayer,
+    TileLayerOptions,
+    TileLayer
+} from "leaflet";
+import { MapProviderEnum, Marker, SharedProps } from "../../typings";
+import { Alert } from "@widgets-resources/piw-utils";
+import classNames from "classnames";
+import { customUrls, getDimensions, mapAttr } from "../utils";
+
+export interface LeafletProps extends SharedProps {
+    mapProvider: MapProviderEnum;
+    attributionControl: boolean;
+}
+
+export const LeafletMap = (props: LeafletProps): ReactElement => {
+    const map = useRef<Map>();
+    const leafletRef = useRef<HTMLDivElement>(null);
+    const defaultCenterLocation: LatLngLiteral = { lat: 51.906688, lng: 4.48837 };
+    const markerGroup = useRef<FeatureGroup>(new FeatureGroup());
+
+    const [validationMessage, setValidationMessage] = useState(props.validationMessage);
+
+    useEffect(() => {
+        createMap();
+
+        return () => {
+            if (map.current) {
+                map.current.off();
+                map.current.remove();
+            }
+        };
+    }, []);
+
+    useEffect(() => {
+        addMarkers();
+    }, [props.locations, props.currentLocation]);
+
+    const createMap = (): void => {
```

Comment:
Same comments applies as for GoogleMaps

Reply:
I applied same things for both, except the add current location because the logic is different